### PR TITLE
Include `AZURE_ENVIRONMENT` parameter for cloud storage docs

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-3458

Adds the `AZURE_ENVIRONMENT` parameter to the cdc and cloud storage documentation for v21.2, v22.1, v22.2

### CODE REVIEW

**Comment by kathancox** [comment]
Hey @adityamaru, could you review this quick docs addition? For context: https://github.com/cockroachdb/cockroach/pull/80800

**Comment by adityamaru** [comment]
The values we accept for `AZURE_ENVIRONMENT` are:

"AZURECHINACLOUD"
"AZUREGERMANCLOUD"
"AZUREPUBLICCLOUD"
"AZUREUSGOVERNMENTCLOUD" 

Azure docs are not very good on what these environments differ in but my read of the code is that depending on the Cloud Environment there are different sets of endpoints that the user can interact with. The link in your PR seems the most appropriate place to link to, but we should probably specify the values we accept in this parameter, and maybe a line about what that means?

**Review by adityamaru — APPROVED** [review]
Thank you!

**Comment by kathancox** [comment]
> The values we accept for `AZURE_ENVIRONMENT` are:
> 
> "AZURECHINACLOUD" "AZUREGERMANCLOUD" "AZUREPUBLICCLOUD" "AZUREUSGOVERNMENTCLOUD"
> 
> Azure docs are not very good on what these environments differ in but my read of the code is that depending on the Cloud Environment there are different sets of endpoints that the user can interact with. The link in your PR seems the most appropriate place to link to, but we should probably specify the values we accept in this parameter, and maybe a line about what that means?

Ah yesss, sorry didn't gather the values from the PR — my mistake. Yes, what to link to is a little tricky, there is a lot of documentation around US Gov, so I have added that here. From the documentation, it looked like the German cloud env is not having any further development against it.

File: v21.2/changefeed-sinks.md [hunk 1]
```diff
@@ -140,10 +140,18 @@ Examples of supported cloud storage sink URIs:
 
 <a name ="cloud-parameters"></a>The following table lists the available parameters for cloud storage sink URIs:
 
-URI Parameter      | Description
--------------------+------------------------------------------------------------------
-`file_size`        | The file will be flushed (i.e., written to the sink) when it exceeds the specified file size. This can be used with the [`WITH resolved` option](create-changefeed.html#options), which flushes on a specified cadence. <br><br>**Default:** `16MB`
-`topic_prefix`     | Adds a prefix to all topic names.<br><br>For example, `CREATE CHANGEFEED FOR TABLE foo INTO 's3://...?topic_prefix=bar_'` would emit rows under the topic `bar_foo` instead of `foo`.
+URI Parameter      | Storage | Description
+-------------------+------------------------+---------------------------
+`AWS_ACCESS_KEY_ID` | AWS | The access key to your AWS account.
```
Comment (andf-crl) [inline v21.2/changefeed-sinks.md:145]: nit: this is usually referred to as the "access key ID" in AWS and third party docs, tossing "ID" in there helps to  differentiate from the similarly-named "secret access key".

File: v22.2/use-cloud-storage-for-bulk-operations.md [hunk 2]
```diff
@@ -32,8 +32,8 @@ The following table provides a list of the parameters supported by each storage
...
-Google Cloud                                                | `gs`        | Bucket name                                      | `AUTH`: `implicit`, or `specified` (default: `specified`); `CREDENTIALS` <br>[`ASSUME_ROLE`](cloud-storage-authentication.html#google-cloud-storage-assume-role) (optional): Pass the [service account name](https://cloud.google.com/iam/docs/understanding-service-accounts) of the service account to assume. <br><br>For more information, see [Authentication - Google Cloud Storage](cloud-storage-authentication.html#google-cloud-storage-specified).
+Azure                                                       | `azure`     | Storage container                                | `AZURE_ACCOUNT_KEY`, `AZURE_ACCOUNT_NAME`<br><br>`AZURE_ENVIRONMENT`: (optional) {% include {{ page.version.version }}/misc/azure-env-param.md %}<br><br>You must [url encode](https://en.wikipedia.org/wiki/Percent-encoding) your Azure account key before authenticating to Azure Storage. For more information, see [Authentication - Azure Storage](cloud-storage-authentication.html#azure-storage-authentication).
```
Comment (andf-crl) [inline v22.2/use-cloud-storage-for-bulk-operations.md:35]: nit: The url encode notice regarding `AZURE_ACCOUNT_NAME` now follows `AZURE_ENVIRONMENT`. What do you think about breaking this up into three bullet points, and adding the "You must" sentence to the `AZURE_ACCOUNT_NAME` entry directly? Something like:

>- AZURE_ACCOUNT_KEY: Your Azure account key. You must [url encode](https://en.wikipedia.org/wiki/Percent-encoding) your Azure account key before authenticating to Azure Storage. For more information, see [Authentication - Azure Storage](https://deploy-preview-16137--cockroachdb-docs.netlify.app/docs/v22.2/cloud-storage-authentication#azure-storage-authentication).
>- AZURE_ACCOUNT_NAME: The name of your Azure account.
>- AZURE_ENVIRONMENT: (optional) {% include {{ page.version.version }}/misc/azure-env-param.md %}

Could definitely skip the bullets to keep with the rest of the table, maybe just three separate lines one for each param?

File: v22.1/use-cloud-storage-for-bulk-operations.md [hunk 3]
```diff
@@ -27,8 +27,8 @@ URLs for the files you want to import must use the format shown below. For examp
...
+Amazon                                                      | `s3`        | Bucket name                                      | `AUTH`: optional `implicit` or `specified` (default: `specified`); `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`<br><br>[`AWS_SESSION_TOKEN`](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html): (optional) For more information, see [Authentication - Amazon S3](#authentication). <br><br>[`S3_STORAGE_CLASS`](#amazon-s3-storage-classes): (optional) Specify the Amazon S3 storage class for created objects. Note that Glacier Flexible Retrieval and Glacier Deep Archive are not compatible with incremental backups. **Default**: `STANDARD`.
+Azure                                                       | `azure`     | Storage container                                | `AZURE_ACCOUNT_KEY`, `AZURE_ACCOUNT_NAME`<br><br>`AZURE_ENVIRONMENT`: (optional) {% include {{ page.version.version }}/misc/azure-env-param.md %}<br><br>You must [url encode](https://en.wikipedia.org/wiki/Percent-encoding) your Azure account key before authenticating to Azure Storage. For more information, see [Authentication - Azure Storage](#authentication).
```
Comment (andf-crl) [inline v22.1/use-cloud-storage-for-bulk-operations.md:31]: Consider my nit regarding breaking this up into bullets -- or even just three separate lines -- below.

**Review by andf-crl — APPROVED** [review]
LGTM, just small optional layout comments. Thanks!

File: v21.2/changefeed-sinks.md (hunk 1 above)
Comment (kathancox) [reply v21.2/changefeed-sinks.md:145]: Yeah, good point!

File: v22.2/use-cloud-storage-for-bulk-operations.md (hunk 2 above)
Comment (kathancox) [reply v22.2/use-cloud-storage-for-bulk-operations.md:35]: I hate the bullets with the difference in font, so following the formatting in the other cells. 👍

Comment (kathancox) [reply v22.2/use-cloud-storage-for-bulk-operations.md:35]: Honestly, I think this table probably needs dismantling into sections soon.

Comment (andf-crl) [reply v22.2/use-cloud-storage-for-bulk-operations.md:35]: Making lists in tables here is always a huge challenge. I think you're probably right. I have tables like this too! 😅

**Pull request merged**